### PR TITLE
Yc 800 801 enable modules

### PR DESCRIPTION
**Pull request by monodo**

### CODE REVIEW

File: permits/models.py [hunk 1]
```diff
@@ -1071,7 +1071,12 @@ class WorksObjectType(models.Model):
     days_before_reminder = models.IntegerField(
         _("Délai de rappel (jours)"), blank=True, null=True
     )
-
+    is_document_management_enabled = models.BooleanField(
+        _("Activer la gestion des documents"), default=False
```
Comment (rbovard) [inline permits/models.py:1075]: `Activer le module de gestion des documents` to be coherent with `Activer le module de publication`?

File: permits/models.py [hunk 2]
```diff
@@ -1071,7 +1071,12 @@ class WorksObjectType(models.Model):
     days_before_reminder = models.IntegerField(
         _("Délai de rappel (jours)"), blank=True, null=True
     )
-
+    is_document_management_enabled = models.BooleanField(
```
Comment (rbovard) [inline permits/models.py:1074]: `document_management_module_enabled`?

File: permits/forms.py [hunk 3]
```diff
@@ -958,9 +962,11 @@ def __init__(self, *args, **kwargs):
             self.fields["status"].choices = tuple(filter2)
 
             if not config.ENABLE_GEOCALENDAR:
-                self.fields["is_public"].widget = forms.HiddenInput()
                 self.fields["shortname"].widget = forms.HiddenInput()
 
+            if not services.is_publication_enabled_for_wots(self.instance):
```
Comment (rbovard) [inline permits/forms.py:967]: Should be in `if not config.ENABLE_GEOCALENDAR` bloc?

File: permits/services.py [hunk 4]
```diff
@@ -1967,3 +1967,26 @@ def get_services_to_notify_mailing_list(permit_request):
...
+def is_publication_enabled_for_wots(permit_request):
+    """ 
+        All WOT must have publication enable in order to allow publication. 
+        We don't want accidental publication. If the application instance has no 
+        geocalendar enabled, the publication is also disabled
+    """
+    return (
+        permit_request.works_object_types.filter(is_publication_enabled=True).count()
+        == permit_request.works_object_types.count()
+    ) and config.ENABLE_GEOCALENDAR
```
Comment (rbovard) [inline permits/services.py:1992]: I dont't know if it's better to put that in this function or when you call this function. (see my previous comment)

File: permits/services.py [hunk 5]
```diff
@@ -1967,3 +1967,26 @@ def get_services_to_notify_mailing_list(permit_request):
...
+
+    return (
+        permit_request.works_object_types.filter(
+            is_document_management_enabled=True
+        ).count()
+        > 0
+    )
+
+
+def is_publication_enabled_for_wots(permit_request):
```
Comment (rbovard) [inline permits/services.py:1983]: `has_publication_module_enabled_for_wots`

**Review by rbovard — CHANGES_REQUESTED** [review]

File: permits/forms.py (hunk 3 above)
Comment (monodo) [reply permits/forms.py:967]: No, the methods `has_publication_enabled_for_wots` now include check for geocalendar activation

File: permits/services.py (hunk 4 above)
Comment (monodo) [reply permits/services.py:1992]: I thinks it's  better to keep the display/hide logic at the same place in the method

File: permits/migrations/0072_enabled_disabled_modules_by_wot.py [hunk 6]
```diff
@@ -12,12 +12,12 @@ class Migration(migrations.Migration):
...
-            name='is_document_management_enabled',
-            field=models.BooleanField(default=False, verbose_name='Activer la gestion des documents'),
+            name='document_enabled',
+            field=models.BooleanField(default=False, verbose_name='Activer le module gestion des documents'),
         ),
         migrations.AddField(
             model_name='worksobjecttype',
-            name='is_publication_enabled',
+            name='publication_enabled',
             field=models.BooleanField(default=False, verbose_name='Activer le module de publication'),
```
Comment (rbovard) [inline permits/migrations/0072_enabled_disabled_modules_by_wot.py:21]: `Activer la gestion de la publication`

File: permits/migrations/0072_enabled_disabled_modules_by_wot.py [hunk 7]
```diff
@@ -12,12 +12,12 @@ class Migration(migrations.Migration):
     operations = [
         migrations.AddField(
             model_name='worksobjecttype',
-            name='is_document_management_enabled',
-            field=models.BooleanField(default=False, verbose_name='Activer la gestion des documents'),
+            name='document_enabled',
+            field=models.BooleanField(default=False, verbose_name='Activer le module gestion des documents'),
```
Comment (rbovard) [inline permits/migrations/0072_enabled_disabled_modules_by_wot.py:16]: `Activer la gestion des documents`

File: permits/templates/permits/permit_request_confirm_inquiry.html [hunk 8]
```diff
@@ -6,7 +6,7 @@
   <h3 class="mt-2">{% trans "Mise en consultation publique" %}</h3>
     <form method="post">
       {% csrf_token %}
-      <p>{% trans "Attention, les documents suivants devienderont public une fois la mise en consultation publique démarré :" %}</p>
+      <p>{% trans "Attention, les documents suivants devienderont public une fois la mise en consultation publique démarrée :" %}</p>
```
Comment (rbovard) [inline permits/templates/permits/permit_request_confirm_inquiry.html:9]: `Attention, les documents suivants **deviendront publics** une fois la mise en consultation publique démarrée`

**Review by rbovard — CHANGES_REQUESTED** [review]

File: permits/templates/permits/permit_request_confirm_inquiry.html (hunk 8 above)
Comment (rbovard) [reply permits/templates/permits/permit_request_confirm_inquiry.html:9]: @monodo You forgot this one

**Review by danickfort — CHANGES_REQUESTED** [review]
`permanent_publication_enabled` is not used anywhere in the code ?

**Comment by monodo** [comment]
https://github.com/yverdon/geocity/blob/yc-800-801-enable-modules/permits/forms.py#L968 here @danickfort

**Pull request merged**